### PR TITLE
update ansible tests and refactor to separate some concerns

### PR DESCRIPTION
**Pull request by bhcleek**
* Fix the provisioner/ansible test that was failing in light of #3149.
* Gofmt provisioner/ansible/provisioner.go
* separate key generation concerns into separate structures and constructors for host signer and user key.

### CODE REVIEW

**Comment by bhcleek** [comment]
Fixes #3173

File: website/source/docs/provisioners/ansible.html.markdown [hunk 1]
```diff
@@ -47,18 +47,17 @@ Required Parameters:
...
+- `ssh_host_key_file` (string) - The SSH key that will be used to run the SSH
+  server on the host machine to forward commands to the target machine. Ansible
+  connects to this server and will validate the identity of the server using
+  the system known_hosts. The default behaviour is to generate and use a one
+  time key, and disable host_key_verification in ansible to allow it to connect
+  to the server 
+
+- `ssh_authorized_key_file` (string) - The SSH public key of the Ansible
+  `ssh_user`. The default behaviour is to generate and use a one time key. If
+  this file is generated the coorisponding private key will be passed via the
```
Comment (rickard-von-essen) [inline website/source/docs/provisioners/ansible.html.markdown]: Spelling

File: website/source/docs/provisioners/ansible.html.markdown [hunk 2]
```diff
@@ -47,18 +47,17 @@ Required Parameters:
...
-  server 
-
-- `ssh_authorized_key_file` (string) - The SSH public key of the
-  Ansible `ssh_user`. The default behaviour is to generate and use a
-  one time key. If this file is generated the coorisponding private
-  key will be passed via the `--private-key` option to Ansible.
+- `ssh_host_key_file` (string) - The SSH key that will be used to run the SSH
+  server on the host machine to forward commands to the target machine. Ansible
+  connects to this server and will validate the identity of the server using
+  the system known_hosts. The default behaviour is to generate and use a one
```
Comment (rickard-von-essen) [inline website/source/docs/provisioners/ansible.html.markdown]: one time => onetime

Occurs  in some more places.

**Comment by bhcleek** [comment]
PTAL

**Comment by bhcleek** [comment]
PTAL

**Comment by rickard-von-essen** [comment]
:+1:

**Comment by rickard-von-essen** [comment]
Could you rebase onto master.

**Comment by bhcleek** [comment]
PTAL

**Comment by cbednarski** [comment]
Looks great, thanks @bhcleek and @rickard-von-essen !

**Pull request merged**